ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -1015,20 +1024,26 @@ getDocsNonInteractive name = do
                     then Left $ NoDocsInIface mod isNameCompiled
                     else Right (Map.lookup name dmap, Map.lookup name amap)
 
+-- | Non-interactive modification of 'GHC.Runtime.Eval.getDocs'.
+--   The interactive paths create problems in ghc-lib builds
+---  and lead to fun errors like "Cannot continue after interface file error".
+getDocsNonInteractive :: HscEnv -> Module -> Name -> IO (Either ErrorMessages (Name, Either GetDocsFailure (Maybe HsDocString, Maybe (Map.Map Int HsDocString))))
+getDocsNonInteractive hsc_env mod name = do
+    ((_warns,errs), res) <- initTypecheckEnv hsc_env mod $ getDocsNonInteractive' name
+    pure $ maybeToEither errs res
+
+
 -- | Non-interactive, batch version of 'GHC.Runtime.Eval.getDocs'.
 getDocsBatch
   :: HscEnv
   -> Module  -- ^ a moudle where the names are in scope
   -> [Name]
   -> IO (Either ErrorMessages (Map.Map Name (Either GetDocsFailure (Maybe HsDocString, Maybe (Map.Map Int HsDocString)))))
   -- ^ Return a 'Map' of 'Name's to 'Either' (no docs messages) (general doc body & arg docs)
-getDocsBatch hsc_env _mod _names = do
```

Comment:
`_names` is a convention for naming semantic meaning of bottoms, or `_` also gets used for gettters. Since these are neither & are normal arguments - using normal name for them.